datadog_checks_dev/datadog_checks/dev/tooling/commands/meta/snmp/generate_traps_db.py

```diff
@@ -0,0 +1,300 @@
+# (C) Datadog, Inc. 2022-present
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+import json
+import os
+from functools import lru_cache
+
+import click
+import yaml
+from pysmi.codegen import JsonCodeGen
+from pysmi.compiler import MibCompiler
+from pysmi.parser import SmiV1CompatParser
+from pysmi.reader import getReadersFromUrls
+from pysmi.searcher import AnyFileSearcher
+from pysmi.writer import FileWriter
+
+from datadog_checks.dev import TempDir
+
+from ...console import CONTEXT_SETTINGS, abort, echo_failure, echo_info, echo_success, echo_warning, set_debug
+from .constants import MIB_SOURCE_URL
+
+# Unique identifiers of traps in json-compiled MIB files.
+NOTIFICATION_TYPE = 'notificationtype'
+ALLOWED_EXTENSIONS_BY_FORMAT = {"json": [".json"], "yaml": [".yml", ".yaml"]}
+
+
+@click.command(
+    context_settings=CONTEXT_SETTINGS,
+    short_help='Generate a traps database that can be used by the '
+    'Datadog Agent for resolving Traps OIDs to readable strings.',
+)
+@click.option(
+    '--mib-sources',
+    '-s',
+    help='Url or a path to a directory containing the dependencies for [mib_files...].'
+    'Traps defined in these files are ignored.',
+)
+@click.option(
+    '--output-dir',
+    '-o',
+    help='Path to a directory where to store the created traps database file per MIB.'
+    'Recommended option, do not use with --output-file',
+    type=click.Path(exists=True, file_okay=False, resolve_path=True),
+)
+@click.option(
+    '--output-file',
+    help='Path to a file to store a compacted version of the traps database file. Do not use with --output-dir',
+    type=click.Path(exists=False, dir_okay=False, resolve_path=True),
+)
+@click.option(
+    '--output-format',
+    type=click.Choice(['yaml', 'json'], case_sensitive=False),
+    default='yaml',
+    help='Use json instead of yaml for the output file(s).',
+)
+@click.option('--debug', '-d', help='Include debug output', is_flag=True)
+@click.argument('mib-files', nargs=-1, required=True, type=click.Path(exists=True, dir_okay=False, resolve_path=True))
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏I think we should be able to use a folder as input too and read all the MIBs from that folder.

Currently, I'm getting:

```
$ ddev meta snmp generate-traps-db --output-dir ./output my-mibs
Error: Invalid value for 'MIB_FILES...': File 'my-mibs' is a directory.
```

Reply:
I think that it is more standard for UNIX cli tool to work like this. Supporting both may be counterintuitive.
For instance let's say you have the following structure:
```
current_dir/
    sub_dir/
```
and you run ```ddev meta snmp generate-traps-db --output-dir ./output current_dir/*```

Because the shell does completion before calling the command, it means that the program will receive `current_dir/sub_dir` as an input which would not be the expectation here.

Other option would be to support only a single directory argument as an input but using a list of files seems more practical for a CLI tool